### DISCUSSION TITLE
OCAPI order_total on basket

### DISCUSSION THREAD

**Discussion by chuan0** [SalesforceCommerceCloud/sfcc-ci#450] [Team Posts]
Trying to use OCAPI to create a basket and submit an order. But I'm getting an OrderTotalNotSet flash(error) 
These are the steps I did:
 1. Create basket using POST /baskets
 2. Add item to basket using POST /baskets/{basket_id}/items
 3. Set shipping address using PUT /baskets/{basket_id}/shipments/{shipment_id}/shipping_address
 4. Set shipping method using PUT /baskets/{basket_id}/shipments/{shipment_id}/shipping_method 
 5. Set payment instrument using POST /baskets/{basket_id}/payment_instruments 

I checked all the endpoints under /baksets I don't see any that calculates the basket. Am I missing something? Or do I have to use a hook and manually calculate the basket in the backend code?

**Comment by saurabh-jain012** [comment]
Call calculate basket hook 'dw.ocapi.shop.basket.calculate' after setting shipping method in afterPUT, it should resolve this.

**Comment by chuan0** [comment]
Thanks for the help. We actually have hooks already, there's a bug in the code that skipped tax calculation and basket calculate.
